pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -334,11 +335,17 @@ func getDomainsForCertBundle(cb hivev1alpha1.CertificateBundleSpec, cd *hivev1al
 	domains := []string{}
 	dLogger := logger.WithValues("CertificateBundle", cb.Name)
 
-	// first check for the special-case default control plane reference
+	// first check for the special-case two default control plane references
```

Comment:
The code looks good, but the comment doesn't make much sense to me. (I know it's not originally your comment, but anything we can do to clear it up might be helpful.)

Maybe something like `// First check if CertificateBundleSpec.Name matches the default control plane name`.

Also, since that's what this block is doing, does it make sense to check for the env var inside this block? That is, do we really need the default ServingCertificate to equal the CertificateBundleSpec name before appending the extra record? What if the default Serving Cert is not equal to the CertificateBundleSpec?

Sorry for the inconvenience. I'm just trying to understand this fully.

Reply:
The reason for this being in the if block is that this extra record is just an alternate name for the cluster api. Using it you can get a cluster with a cert for the cluster api of `api.cluster.domain` and `custom.cluster.domain`. But if we put it out of the if block it would be applied to ever other cert for the cluster.

Having a cluster with multiple certificates for custom ingress objects is not common but is supported by Hive. By default Hive for OSD asks for certs with `api.cluster`, `*.apps.cluster`. But the cluster deployment object may ask for additional certs and domain records which can be applied to Ingresses. Placing this option inside the if statements prevents the new option from being applied to these certs also.

That comment could totally be improved. +1 on that.